test/test_interpreter.cpp

```diff
@@ -48,3 +48,232 @@ TEST_SUITE("extract_filename")
         REQUIRE(argc == 2);
     }
 }
+
+TEST_SUITE("should_print_version")
+{
+    // This test case checks if the function `should_print_version` correctly identifies
+    // when the "--version" argument is passed. It sets up a scenario where "--version"
+    // is one of the command line arguments and checks if the function returns true.
+    TEST_CASE("should_print_version_with_version_arg")
+    {
+        char arg1[] = "program_name";
+        char arg2[] = "--version";
+        char* argv[] = {arg1, arg2};
+        int argc = 2;
+
+        bool result = xcpp::should_print_version(argc, argv);
+
+        REQUIRE(result == true);
+    }
+
+    // This test case checks if the function `should_print_version` correctly identifies
+    // when the "--version" argument is not passed. It sets up a scenario where "--version"
+    // is not one of the command line arguments and checks if the function returns false.
+    TEST_CASE("should_print_version_without_version_arg")
+    {
+        char arg1[] = "program_name";
+        char arg2[] = "-f";
+        char arg3[] = "filename";
+        char* argv[] = {arg1, arg2, arg3};
+        int argc = 3;
+
+        bool result = xcpp::should_print_version(argc, argv);
+
+        REQUIRE(result == false);
+    }
+}
+
+TEST_SUITE("build_interpreter")
+{
+    // This test case checks if the function `build_interpreter` returns a non-null pointer
+    // when valid arguments are passed. It sets up a scenario with valid command line arguments
+    // and checks if the function returns a non-null pointer.
+    TEST_CASE("build_interpreter_pointer_not_null")
+    {
+        char arg1[] = "program_name";
+        char arg2[] = "-option1";
+        char* argv[] = {arg1, arg2};
+        int argc = 2;
+
+        interpreter_ptr interp_ptr = xcpp::build_interpreter(argc, argv);
+
+        REQUIRE(interp_ptr != nullptr);
+    }
+}
+
+TEST_SUITE("is_match_magics_manager")
```

Comment:
@brauliorivas if I understand correctly you can improve this test to check `xmagics_manager::apply` with line magic instead. it would std::regex_search %python same as is_match but also check if the magic can be successfully applied(and possibly record any breaking of the magic commands)

Reply:
alright! I'll change it

Reply:
Hey @maximusron I'm working on this test case as you said but I'm having trouble. If you have time, some help would be handy!